realm/kotlin-extensions/src/androidTest/kotlin/io/realm/CoroutineTests.kt

```diff
@@ -0,0 +1,147 @@
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import java.util.concurrent.CountDownLatch
+import kotlin.test.*
+
+class CoroutineTests {
+
+    @Suppress("MemberVisibilityCanPrivate")
+    @Rule
+    @JvmField
+    val configFactory = TestRealmConfigurationFactory()
+
+    private lateinit var testDispatcher: TestCoroutineDispatcher
+    private lateinit var testScope: TestCoroutineScope
+    private lateinit var realm: Realm
+
+    private val looperThread = BlockingLooperThread()
+
+    @Before
+    fun setUp() {
+        realm = Realm.getInstance(configFactory.createConfiguration())
+
+        with(Realm.getDefaultInstance()) {
+            executeTransaction { it.deleteAll() }
+            close()
+        }
+
+        testDispatcher = TestCoroutineDispatcher()
+        testScope = TestCoroutineScope(testDispatcher)
+    }
+
+    @After
+    fun tearDown() {
+        realm.executeTransaction { it.deleteAll() }
+        realm.close()
+    }
+
+    @Test
+    fun toFlow_emittedOnCollect() {
+        Realm.getDefaultInstance().use { defaultRealm ->
+            defaultRealm.executeTransactionAsync {
+                it.createObject(SimpleClass::class.java).name = "Foo"
+                it.createObject(SimpleClass::class.java).name = "Bar"
+            }
+        }
+
+        val countDownLatch = CountDownLatch(1)
+
+        // TODO check this out for better testing: https://proandroiddev.com/from-rxjava-to-kotlin-flow-testing-42f1641d8433
+        val scope = CoroutineScope(Dispatchers.Main)
+        scope.launch {
+            Realm.getDefaultInstance()
+                    .where(SimpleClass::class.java)
+                    .findAll()
+                    .toFlow()
+                    .onEach { flowResults ->
+                        assertTrue(flowResults.isFrozen)
+                        if (flowResults.size == 2) {
+                            scope.cancel()
+                        }
+                    }.onCompletion {
+                        countDownLatch.countDown()
+                    }.launchIn(scope)
+        }
+        countDownLatch.await()
+    }
+
+    @Test
+    fun toFlow_throwsDueToThreadViolation() {
+        val countDownLatch = CountDownLatch(1)
+
+        // Get results from the test thread
+        val findAll = Realm.getDefaultInstance()
+                .where(SimpleClass::class.java)
+                .findAll()
+
+        CoroutineScope(Dispatchers.Main).launch {
+            assertFailsWith<IllegalStateException> {
+                // Now we are on the main thread, which means crash
+                findAll.toFlow().collect()
+                fail("toFlow() must be called from the thread that retrieved the results!")
+            }
+            countDownLatch.countDown()
+        }
+        countDownLatch.await()
+    }
+
+    @Test
+    fun executeTransactionAwait() {
+        testScope.runBlockingTest {
+            Realm.getDefaultInstance().use { defaultRealm ->
```

Comment:
Use `Realm.getInstance(realm.configuration)` if you need to use a configuration from the existing Realm or save the configuration in an test class property so you can do `Realm.getInstance(config)`